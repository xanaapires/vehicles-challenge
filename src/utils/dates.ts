import moment from "moment";

export const getRelativeTime = (start: Date) => {
  const now = moment();
  const target = moment(start);

  const isPast = target.isBefore(now);
  const duration = moment.duration(Math.abs(target.diff(now)));

  const days = Math.floor(duration.asDays());
  const hours = duration.hours();

  return isPast
    ? `${days} day(s) and ${hours} hour(s) ago`
    : `in ${days} day(s) and ${hours} hour(s)`;
};
