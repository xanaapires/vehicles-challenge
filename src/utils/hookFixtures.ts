import { vehiclesData } from "../assets/vehiclesData";

export const defaultStateMock = {
  vehiclesToShow: vehiclesData,
  filteredVehicles: vehiclesData,
  filters: {
    make: "",
    model: "",
    startBid: "",
    endBid: "",
    favouritesOnly: false,
  },
  applyFilter: () => {},
  availableMakes: [
    {
      label: "All",
      value: "all",
    },
    {
      label: "Audi",
      value: "Audi",
    },
    {
      label: "BMW",
      value: "BMW",
    },
    {
      label: "Citroen",
      value: "Citroen",
    },
    {
      label: "Ford",
      value: "Ford",
    },
    {
      label: "Mercedes-Benz",
      value: "Mercedes-Benz",
    },
    {
      label: "Toyota",
      value: "Toyota",
    },
    {
      label: "Volkswagen",
      value: "Volkswagen",
    },
    {
      label: "Volvo",
      value: "Volvo",
    },
  ],
  availableModels: [],
  bidValues: [],
  getVehicleById: () => {},
  toggleFavourite: () => {},
};
