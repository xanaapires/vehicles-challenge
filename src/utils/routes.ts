export const ROUTES = {
  LISTING: "/",
  VEHICLE_DETAILS: "/vehicle/:vehicleId",
};
