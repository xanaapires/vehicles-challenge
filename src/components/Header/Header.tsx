import React from "react";
import { Box } from "@mui/material";
import icon from "../../assets/icon.jpeg";
import { useNavigate } from "react-router-dom";
import { ROUTES } from "../../utils/routes";

const Header = () => {
  const navigate = useNavigate();

  const handleGoHome = () => {
    navigate(ROUTES.LISTING);
  };

  return (
    <Box display="flex" alignItems="center" padding="24px">
      <img
        data-testid="cba-logo"
        src={icon}
        alt="Icon"
        style={{ height: "50px", cursor: "pointer" }}
        onClick={handleGoHome}
      />
    </Box>
  );
};

export default Header;
