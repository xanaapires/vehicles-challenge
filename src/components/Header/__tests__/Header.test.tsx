import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Header from "../Header";
import { MemoryRouter } from "react-router-dom";
import { ROUTES } from "../../../utils/routes";

const mockedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedNavigate,
}));

const renderComponent = () => {
  return render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>,
  );
};

describe("Header", () => {
  test("Renders header correctly with Logo", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
    expect(screen.getByTestId("cba-logo")).toBeVisible();
    fireEvent.click(screen.getByTestId("cba-logo"));

    expect(mockedNavigate).toHaveBeenCalledWith(ROUTES.LISTING);
  });
});
