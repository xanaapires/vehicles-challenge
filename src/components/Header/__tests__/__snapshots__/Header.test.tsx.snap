// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders header correctly with Logo 1`] = `
<div>
  <div
    class="MuiBox-root css-43yz9u"
  >
    <img
      alt="Icon"
      data-testid="cba-logo"
      src="icon.jpeg"
      style="height: 50px; cursor: pointer;"
    />
  </div>
</div>
`;
